src/tooltip_test.js

```diff
@@ -0,0 +1,99 @@
+/*global CustomEvent*/
+if (typeof process !== "undefined") {
+    require("./test/mockdom");
+}
+
+"use strict";
+
+var ace = require("./ace");
+var assert = require("./test/assertions");
+var HoverTooltip = require("./tooltip").HoverTooltip;
+var editor, docTooltip;
+module.exports = {
+    setUp: function() {
+        docTooltip = new HoverTooltip();
+        editor = ace.edit(null, {
+            value: "Hello empty world\n"
+        });
+        document.body.appendChild(editor.container);
+        editor.container.style.height = "200px";
+        editor.container.style.width = "300px";
+    },
+    tearDown: function() {
+        editor.destroy();
+        docTooltip.destroy();
+        editor = docTooltip = null;
+    },
+   "test: show doc tooltip" : function(next) {
+        docTooltip.addToEditor(editor);
+        
+        docTooltip.setDataProvider(function(e, editor) {
+            let session = editor.session;
+            let docPos = e.getDocumentPosition();
+            
+            var range = session.getWordRange(docPos.row, docPos.column);
+
+            var domNode = document.createElement("span");
+            domNode.textContent = "tooltip " + range;
+            domNode.className = "doc-tooltip";
+            docTooltip.showForRange(editor, range, domNode, e);
+        });
+        
+        
+        editor.resize(true);
+        docTooltip.idleTime = 3;
+        mouse("move", {row: 0, column: 1});
+        setTimeout(function() {
+            var nodes = document.querySelectorAll(".doc-tooltip");
+            assert.equal(nodes.length, 1);
+            assert.equal(nodes[0].textContent, "tooltip Range: [0/0] -> [0/5]");
+            assert.equal(docTooltip.$element.style.display, "block");
+            mouse("move", {row: 0, column: 9});
+            assert.equal(docTooltip.$element.style.display, "none");
+            setTimeout(function() {
+                assert.equal(docTooltip.$element.textContent, "tooltip Range: [0/6] -> [0/11]");
+                assert.equal(docTooltip.$element.style.display, "block");
+                mouse("down", docTooltip.$element, {button: 0});
+                mouse("up", docTooltip.$element, {button: 0});
+                assert.equal(docTooltip.$element.style.display, "block");
+                mouse("down", {row: 0, column: 8}, {button: 0});
+                    // console.log(editor.$mouseHandler.isMousePressed)
```

Comment:
Can we remove this commented out line here.

Reply:
Done